examples/tv-casting-app/linux/simple-app-helper.cpp

```diff
@@ -256,23 +268,49 @@ void ConnectionHandler(CHIP_ERROR err, matter::casting::core::CastingPlayer * ca
             "simple-app-helper.cpp::ConnectionHandler(): Failed to connect to CastingPlayer (ID: %s) with err %" CHIP_ERROR_FORMAT,
             targetCastingPlayer->GetId(), err.Format()));
 
-    ChipLogProgress(AppServer, "simple-app-helper.cpp::ConnectionHandler(): Successfully connected to CastingPlayer (ID: %s)",
-                    castingPlayer->GetId());
-    ChipLogProgress(AppServer,
-                    "simple-app-helper.cpp::ConnectionHandler(): Triggering demo interactions with CastingPlayer (ID: %s)",
-                    castingPlayer->GetId());
+    if (gCommissionerGeneratedPasscodeFlowRunning)
+    {
+        ChipLogProgress(AppServer,
+                        "simple-app-helper.cpp::ConnectionHandler(): Successfully connected to CastingPlayer (ID: %s) using "
+                        "Commissioner-Generated passcode",
+                        castingPlayer->GetId());
+        ChipLogProgress(AppServer, "simple-app-helper.cpp::ConnectionHandler(): Desired Endpoint ID for demo interactions: 1");
+    }
+    else
+    {
+        ChipLogProgress(AppServer, "simple-app-helper.cpp::ConnectionHandler(): Successfully connected to CastingPlayer (ID: %s)",
+                        castingPlayer->GetId());
+        ChipLogProgress(AppServer,
+                        "simple-app-helper.cpp::ConnectionHandler(): Desired Endpoint Vendor ID for demo interactions: %d",
+                        kDesiredEndpointVendorId);
+    }
 
+    ChipLogProgress(AppServer, "simple-app-helper.cpp::ConnectionHandler(): Getting endpoints avaiable for demo interactions");
     std::vector<matter::casting::memory::Strong<matter::casting::core::Endpoint>> endpoints = castingPlayer->GetEndpoints();
+    LogEndpointsDetails(endpoints);
+
     // Find the desired Endpoint and auto-trigger some Matter Casting demo interactions
     auto it = std::find_if(endpoints.begin(), endpoints.end(),
                            [](const matter::casting::memory::Strong<matter::casting::core::Endpoint> & endpoint) {
-                               return endpoint->GetVendorId() == kDesiredEndpointVendorId;
+                               if (gCommissionerGeneratedPasscodeFlowRunning)
+                               {
+                                   // Since the tv-app with Vendor ID 1111 does not implement the endpoint with Vendor ID 65521 we
+                                   // will (temporarily) use the endpoint with ID == 1 for demo interactions. See
+                                   // connectedhomeip/examples/tv-app/tv-common/include/AppTv.h.
+                                   return endpoint->GetId() == 1;
```

Comment:
add a constant for 1 since we have a cohnstant for the other value?

The comment says that tv app with vendor id 1111, however I don't see that check here. Should we check for GetVendorId == 1111? Generally asking why the comment about VendorID is here ... also the comment is a bit unclear to me: what is the tv app vendorid and what is the endpoint vendorid? Which one is 1111 and which one is 65521?

Reply:
Add in a constant in the next rev.

To demonstrate the Commissioner-Generated passcode commissioning flow we needed to connect to the Target Content Application with Vendor ID 1111, which doesn't implement the AccountLogin cluster (account login obtains the commissionee passcode form a backend server).

The example tv-app simulates 4 content apps installed on a TV device (see https://github.com/pgregorr-amazon/connectedhomeip/blob/master/examples/tv-app/tv-common/include/AppTv.h#L143-L147). Each content app has a Vendor ID defined (1, 65521, 9050 and 1111). After commissioning with a specific content app, the content app reports the Endpoints available for interactions. Each Endpoint has a Endpoint ID and Endpoint Vendor ID attributes.